custom-metrics-stackdriver-adapter/pkg/adapter/translator/query_builder.go

```diff
@@ -91,64 +91,70 @@ type Translator struct {
+func (pc podValues) isPodValuesValid() bool {
 	return !(len(pc.podNames) > 0 && pc.pods != nil && len(pc.pods.Items) > 0)
 }
 
-// isPodsEmpty checks if pods container is empty
-func (pc podsContainer) isPodsContainerEmpty() bool {
+// isPodValuesEmpty checks if pods container is empty
+func (pc podValues) isPodValuesEmpty() bool {
 	return len(pc.podNames) == 0 && (pc.pods == nil || len(pc.pods.Items) == 0)
 }
 
-// getPodNames gets pod names from podsContainer is any provided
-func (pc podsContainer) getPodNames() []string {
-	if pc.pods != nil {
-		podNames := []string{}
-		for _, item := range pc.pods.Items {
-			podNames = append(podNames, fmt.Sprintf("%q", item.GetName()))
-		}
-		return podNames
+// getPodNames gets pod names from podValues is any provided
+func (pc podValues) getPodNames() []string {
+	if pc.pods == nil {
+		return pc.podNames
 	}
-	return pc.podNames
+	podNames := make([]string, len(pc.pods.Items))
+	for i, item := range pc.pods.Items {
+		podNames[i] = fmt.Sprintf("%q", item.GetName())
+	}
+	return podNames
 }
 
-// getPodIds gets pod ids from podsContainer if any provided
-func (pc podsContainer) getPodIds() []string {
-	if pc.pods != nil {
-		podIds := []string{}
-		for _, item := range pc.pods.Items {
-			podIds = append(podIds, fmt.Sprintf("%q", item.GetUID()))
-		}
-		return podIds
+// getPodIDs gets pod ids from podValues if any provided
+func (pc podValues) getPodIDs() []string {
+	if pc.pods == nil {
+		return []string{}
 	}
-	return []string{}
+	podIDs := make([]string, len(pc.pods.Items))
+	for i, item := range pc.pods.Items {
+		podIDs[i] = fmt.Sprintf("%q", item.GetUID())
+	}
+	return podIDs
 }
 
-// nodesContainer is a helper struct to hold nodes values
-type nodesContainer struct {
+// nodeValues is a helper struct to hold nodes values
+type nodeValues struct {
 	nodes     *v1.NodeList
 	nodeNames []string
 }
 
-// isNodesContainerValid checks if nodeContainer is valid to be used
-func (nc nodesContainer) isNodesContainerValid() bool {
+// isNodeValuesValid checks if nodeContainer is valid to be used
+//
+// this only happens when the both nodes & nodeNames are provided,
+// for example, when WithNodes() and WithNodeNames() are both used in QueryBuilder
+func (nc nodeValues) isNodeValuesValid() bool {
 	return !(len(nc.nodeNames) > 0 && nc.nodes != nil && len(nc.nodes.Items) > 0)
 }
 
-// isNodesContainerEmpty checks if nodes container is empty
-func (nc nodesContainer) isNodesContainerEmpty() bool {
+// isNodeValuesEmpty checks if nodes container is empty
+func (nc nodeValues) isNodeValuesEmpty() bool {
 	return len(nc.nodeNames) == 0 && (nc.nodes == nil || len(nc.nodes.Items) == 0)
 }
 
 // getNodeNames gets node names from nodeContainer if any provided
-func (nc nodesContainer) getNodeNames() []string {
+func (nc nodeValues) getNodeNames() []string {
 	if nc.nodes != nil {
 		nodeNames := []string{}
 		for _, item := range nc.nodes.Items {
```

Comment:
And why are you not quoting the node names too?

Reply:
actually I think I misunderstood your question, pod names are quoted and node names are not quoted, that's because they have different usages for node names and pod names.

for pod names: `one_of("pod1", "pod2")`
for node names: `monitoring.regex.full_match("^(node1|node2)$")`

for regex method, it doesn't have the filtering number constraint, but it's less readable and hard to maintain, so I didn't change the existing usage implementation for filtering pods. If we encounter some usecases that requires to remove the 100 pod filtering constraint in `one_of`, then we can switch to use regex method.

Reply:
Okay makes sense, can you clarify that in the function name? Like getQuotedPodNames?
Or do the quoting when you add them to the query. Either way works for me, whatever you prefer.

Reply:
sure